src/features/agenda/screens/AgendaWeekScreen.tsx

```diff
@@ -85,15 +85,21 @@ export const AgendaWeekScreen = ({ navigation, route }: Props) => {
     return weekData?.data?.flatMap(week => week.items) ?? [];
   }, [weekData?.data]);
 
+  const filteredCalendarData = useHideEventFilter(calendarData);
+
   const calendarMax = useMemo(() => {
     return (
       calendarData.reduce((max, item) => {
         return item.start.weekday > max.start.weekday ? item : max;
       }, calendarData[0]) ?? null
     );
   }, [calendarData]);
-
-  const filteredCalendarData = useHideEventFilter(calendarData);
+  const lenCalendar = useMemo(() => {
+    if (calendarMax && (calendarMax.start.weekday as WeekNum) > 5) {
+      return calendarMax.start.weekday as WeekNum;
+    }
+    return 5;
+  }, [calendarMax]);
```

Comment:
```suggestion

  const weekLength = useMemo(() => {
    if (calendarMax && calendarMax.start.weekday > 5) {
      return calendarMax.start.weekday as WeekNum;
    }
    return 5;
  }, [calendarMax]);
```

Reply:
(changes:
- added a line before the definition
- used a slightly clearer name
- removed the type from the check)